### ISSUE TITLE
Stepper not sync with step content

### ISSUE THREAD

**Issue by GlorianB** [UgnisSoftware/react-spreadsheet-import#171]
#### Description
When the `initialStepState` props is different from `{ type: StepType.upload }`, the stepper is not sync with the content of the step

#### Screenshots
Screenshot 1:
![Capture1](https://github.com/UgnisSoftware/react-spreadsheet-import/assets/50000915/554528e7-acc8-465f-bd54-3336af7d2bef)

Screenshot 2:
![Capture2](https://github.com/UgnisSoftware/react-spreadsheet-import/assets/50000915/d4b1f744-3ee6-4d65-a9ac-84305ea4d0d1)

#### To Reproduce
Code used for screenshot  1:

```
      <ReactSpreadsheetImport
        initialStepState={{
          type: StepType.selectHeader,
          data: [
            ["Josh", "2"],
            ["Charlie", "3"],
            ["Lena", "50"],
          ],
        }}
        isOpen
        onClose={() => {}}
        onSubmit={(data) => {}}
        fields={[
          { key: "name", label: "Nom", fieldType: { type: "input" } },
          { key: "age", label: "Age", fieldType: { type: "input" } },
        ]}
      />
```

Code used for screenshot 2:
```
      <ReactSpreadsheetImport
        initialStepState={{
          type: StepType.matchColumns,
          data: [
            ["Josh", "2"],
            ["Charlie", "3"],
            ["Lena", "50"],
          ],
          headerValues: ["name", "age"],
        }}
        isOpen
        onClose={() => {}}
        onSubmit={(data) => {}}
        fields={[
          { key: "name", label: "Nom", fieldType: { type: "input" } },
          { key: "age", label: "Age", fieldType: { type: "input" } },
        ]}
      />
```

And then click on `Next` button.

**Comment by masiulis** [comment]
Released in 3.0.1!

**Comment by agnar-nomad** [comment]
I am using version 4.6.1 and am still seeing the exact same thing.
I start at StepType.selectHeader, and the indicator is lagging one step.
